packages/plugin-ext/src/main/browser/tasks-main.ts

```diff
@@ -23,18 +23,45 @@ import { RPCProtocol } from '../../api/rpc-protocol';
 import { DisposableCollection } from '@theia/core';
 import { TaskProviderRegistry, TaskResolverRegistry, TaskProvider, TaskResolver } from '@theia/task/lib/browser/task-contribution';
 import { interfaces } from 'inversify';
+import { WorkspaceService } from '@theia/workspace/lib/browser/workspace-service';
+import { TaskInfo } from '@theia/task/lib/common/task-protocol';
+import { TaskWatcher } from '@theia/task/lib/common/task-watcher';
+import { TaskService } from '@theia/task/lib/browser/task-service';
 
 export class TasksMainImpl implements TasksMain {
+    private workspaceRootUri: string | undefined = undefined;
 
     private readonly proxy: TasksExt;
     private readonly disposables = new Map<number, monaco.IDisposable>();
     private readonly taskProviderRegistry: TaskProviderRegistry;
     private readonly taskResolverRegistry: TaskResolverRegistry;
+    private readonly taskWatcher: TaskWatcher;
+    private readonly taskService: TaskService;
+    private readonly workspaceService: WorkspaceService;
 
     constructor(rpc: RPCProtocol, container: interfaces.Container, ) {
         this.proxy = rpc.getProxy(MAIN_RPC_CONTEXT.TASKS_EXT);
         this.taskProviderRegistry = container.get(TaskProviderRegistry);
         this.taskResolverRegistry = container.get(TaskResolverRegistry);
+        this.workspaceService = container.get(WorkspaceService);
+        this.taskWatcher = container.get(TaskWatcher);
+        this.taskService = container.get(TaskService);
+
+        this.workspaceService.roots.then(async roots => {
```

Comment:
hello, where is async part in this method ? (as there is async keyword)

Reply:
@benoitf fixed, thank you!